cpp/ql/test/query-tests/Security/CWE/CWE-193/InvalidPointerDeref.expected

```diff
@@ -340,6 +364,10 @@ subpaths
 | test.cpp:254:9:254:16 | ... = ... | test.cpp:248:24:248:30 | call to realloc | test.cpp:254:9:254:16 | ... = ... | This write might be out of bounds, as the pointer might be equal to $@ + $@. | test.cpp:248:24:248:30 | call to realloc | call to realloc | test.cpp:254:11:254:11 | i | i |
 | test.cpp:264:13:264:14 | * ... | test.cpp:260:13:260:24 | new[] | test.cpp:264:13:264:14 | * ... | This read might be out of bounds, as the pointer might be equal to $@ + $@. | test.cpp:260:13:260:24 | new[] | new[] | test.cpp:261:19:261:21 | len | len |
 | test.cpp:274:5:274:10 | ... = ... | test.cpp:270:13:270:24 | new[] | test.cpp:274:5:274:10 | ... = ... | This write might be out of bounds, as the pointer might be equal to $@ + $@. | test.cpp:270:13:270:24 | new[] | new[] | test.cpp:271:19:271:21 | len | len |
+| test.cpp:319:5:319:21 | ... = ... | test.cpp:313:14:313:27 | new[] | test.cpp:319:5:319:21 | ... = ... | This write might be out of bounds, as the pointer might be equal to $@ + $@. | test.cpp:313:14:313:27 | new[] | new[] | test.cpp:314:20:314:23 | size | size |
```

Comment:
These look like FPs. @MathiasVP what do you think of them?

I'm hoping they've appeared because we're extracting something that lets us catch these, rather than because we're no longer extracting something that lets us filter them out.

Reply:
It's neither, really. It's one of those changes to value categories that we've seen multiple times in this upgrade. It changes the IR from for `*--(*current) = 0;`:
```cpp
10|     r10_1(char)           = Constant[0]              :
10|     r10_2(glval<char **>) = VariableAddress[current] :
10|     r10_3(char **)        = Load[current]            : &:r10_2, m5_4
10|     r10_4(glval<char *>)  = CopyValue                : r10_3
10|     r10_5(char *)         = Load[?]                  : &:r10_4, m8_1
10|     r10_6(int)            = Constant[1]              :
10|     r10_7(char *)         = PointerSub[1]            : r10_5, r10_6
10|     m10_8(char *)         = Store[?]                 : &:r10_4, r10_7
10|     r10_9(glval<char>)    = CopyValue                : r10_7
10|     m10_10(char)          = Store[?]                 : &:r10_9, r10_1
```
to
```cpp
10|     r10_1(char)           = Constant[0]              :
10|     r10_2(glval<char **>) = VariableAddress[current] :
10|     r10_3(char **)        = Load[current]            : &:r10_2, m5_4
10|     r10_4(glval<char *>)  = CopyValue                : r10_3
10|     r10_5(char *)         = Load[?]                  : &:r10_4, m8_1
10|     r10_6(int)            = Constant[1]              :
10|     r10_7(char *)         = PointerSub[1]            : r10_5, r10_6
10|     m10_8(char *)         = Store[?]                 : &:r10_4, r10_7
10|     r10_9(char *)         = Load[?]                  : &:r10_4, m10_8
10|     r10_10(glval<char>)   = CopyValue                : r10_9
10|     m10_11(char)          = Store[?]                 : &:r10_10, r10_1
```
this _exact_ pattern was actually why I created https://github.com/github/codeql/pull/13965. It used to fix the frontend upgrade branch, but for some reason it doesn't anymore 🤔

Reply:
This seems a regression that we need to look at before we merge this.

Reply:
The fix is here: https://github.com/github/codeql/pull/14376. Looks like https://github.com/github/codeql/pull/14008 introduced the regression.